biomage/unstage/unstage.py

```diff
@@ -3,21 +3,151 @@
 import boto3
 import json
 import base64
+import re
 from PyInquirer import prompt
 from github import Github
 
 
-def check_if_exists(org, sandboxid):
-    url = f"https://raw.githubusercontent.com/{org}/iac/master/releases/staging/{sandboxid}.yaml"  # noqa: E501
+def check_if_exists(org, sandbox_id):
+    url = f"https://raw.githubusercontent.com/{org}/iac/master/releases/staging/{sandbox_id}.yaml"  # noqa: E501
 
     s = requests.Session()
     r = s.get(url)
 
     return 200 <= r.status_code < 300
 
 
+def remove_staging_resources(sandbox_id):
+    # Remove staging records
+    dynamodb = boto3.client("dynamodb")
+
+    staged_experiments = dynamodb.scan(
+        TableName="experiments-staging",
+        ProjectionExpression="experimentId",
+        FilterExpression="begins_with(experimentId, :sandbox_id)",
+        ExpressionAttributeValues={":sandbox_id": {"S": sandbox_id}},
+    )
+
+    staged_experiments = [
+        experiment_id["experimentId"]["S"]
+        for experiment_id in staged_experiments.get("Items")
+    ]
+
+    if len(staged_experiments) == 0:
+        return
+
+    staging_tables = [
+        table
+        for table in dynamodb.list_tables().get("TableNames")
+        if re.match(".*-staging", table)
+    ]
+
+    for table in staging_tables:
+
+        records_to_delete = {}
+
+        # Deleting records from DynamoDB requires the primaryKey.
+        # if the table's PK is made up of partitionKey & sortKey,
+        # Then search with partitionKey and iterate through all the records
+        key_schema = (
+            dynamodb.describe_table(TableName=table).get("Table").get("KeySchema")
+        )
+
+        if len(key_schema) > 1:
+            sort_key = key_schema[1].get("AttributeName")
+            for experiment_id in staged_experiments:
+
+                # Query table for all keys
+                items_to_delete = dynamodb.query(
+                    TableName=table,
+                    ProjectionExpression=f"experimentId, {sort_key}",
+                    KeyConditionExpression="experimentId = :experiment_id",
+                    ExpressionAttributeValues={":experiment_id": {"S": experiment_id}},
```

Comment:
this is almost identical to this bit here: https://github.com/biomage-ltd/biomage-utils/pull/13/files#diff-8c6661b6c586d57473339631eed9bfb4c59815aa848b8dced10ed0de1b790a50R505 from the stage experiment, you might want to consider extracting this logic into a dynamodb class module, have that functionality there and in here and in staging script just import and use it. You might want to consider the same for s3 as well to avoid this code duplication.